### PR TITLE
bpo-35471: Remove the macpath module

### PR DESCRIPTION
**Pull request by vstinner**
Python 2.4 dropped MacOS 9 support. The macpath module was deprecated
in Python 3.7. This change removes it.

https://bugs.python.org/issue35471

### CODE REVIEW

File: Lib/posixpath.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 this module as os.path.  The "os.path" name is an alias for this
 module on Posix systems; on other systems (e.g. Mac, Windows),
```
Comment (serhiy-storchaka) [inline Lib/posixpath.py]:
```suggestion
module on Posix systems; on other systems (e.g. Windows),
```

File: PCbuild/lib.pyproj [hunk 2]
```diff
@@ -1816,4 +1814,4 @@
     <Folder Include="xml\sax" />
   </ItemGroup>
   <Import Project="$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)\Python Tools\Microsoft.PythonTools.targets" />
-</Project>
\ No newline at end of file
+</Project>
```
Comment (serhiy-storchaka) [inline PCbuild/lib.pyproj:1817]: Unrelated change.

File: Tools/freeze/freeze.py [hunk 3]
```diff
@@ -124,7 +124,7 @@ def main():
 
     # default the exclude list for each platform
     if win: exclude = exclude + [
-        'dos', 'dospath', 'mac', 'macpath', 'macfs', 'MACFS', 'posix', ]
```
Comment (serhiy-storchaka) [inline Tools/freeze/freeze.py:127]: 'dospath'? Seriously? :wink:

File: Doc/library/os.path.rst [hunk 4]
```diff
@@ -5,8 +5,7 @@
    :synopsis: Operations on pathnames.
 
 **Source code:** :source:`Lib/posixpath.py` (for POSIX),
```
Comment (serhiy-storchaka) [inline Doc/library/os.path.rst]:
```suggestion
**Source code:** :source:`Lib/posixpath.py` (for POSIX) and
```

File: Tools/freeze/freeze.py (hunk 3 above)
Comment (vstinner) [reply Tools/freeze/freeze.py:127]: I don't know any of these modules (except of "posix"), so I chose to not touch this old code...

File: PCbuild/lib.pyproj (hunk 2 above)
Comment (vstinner) [reply PCbuild/lib.pyproj:1817]: It seems like my text editor (vim) likes to add newline. Well, this change shouldn't hurt :-)

**Comment by vstinner** [comment]
@serhiy-storchaka: I made two changes, see my second commit.

**Review by ned-deily — APPROVED** [review]
LGTM, thanks

**Comment by vstinner** [comment]
@ronaldoussoren wrote "I'm in favour of removing macpath.":
https://bugs.python.org/issue35471#msg331702
